parsers/deathloopvaccine.py

```diff
@@ -0,0 +1,286 @@
+        self._death_list = list()
+
+        # flag indicating whether the "process killer" gun is armed
+        self._kill_armed = True
+
+    def reset(self) -> None:
+        """
+        Utility function to clear the death_list and reset the armed flag
+
+        Returns:
+            None:
+        """
+        self._death_list.clear()
+        self._kill_armed = True
+
+    # main parsing logic here
+    def parse(self, timestamp: datetime, text: str) -> None:
+        """
+        Parse a single line from the logfile
+
+        Args:
+            timestamp: A datetime.datetime object, created from the timestamp text of the raw logfile line
+            text: The text following the everquest timestamp
+
+        Returns:
+            None:
+        """
+
+        self.check_for_death(timestamp, text)
+        self.check_not_afk(timestamp, text)
+        self.deathloop_response()
+
+    def check_for_death(self, timestamp: datetime, text: str) -> None:
+        """
+        check for indications the player just died, and if we find it,
+        save the message for later processing
+
+        Args:
+            timestamp: A datetime.datetime object, created from the timestamp text of the raw logfile line
+            text: The text following the everquest timestamp
+
+        Returns:
+            None:
+        """
+
+        # reconstruct the full logfile line
+        # this is a bit counter-intuitive, but the rest of the logic in this function was
+        # developed assuming the line was the full line, including the time stamp
+        line = f'[{timestamp.strftime("%a %b %d %H:%M:%S %Y")}] ' + text
+
+        # cut off the leading date-time stamp info
+        trunc_line = line[27:]
+
+        # does this line contain a death message
+        slain_regexp = r'^You have been slain'
+        m = re.match(slain_regexp, trunc_line)
+        if m:
+            # add this message to the list of death messages
+            self._death_list.append(line)
+            starprint(f'DeathLoopVaccine:  Death count = {len(self._death_list)}')
+
+        # a way to test - send a tell to death_loop
+        slain_regexp = r'^death_loop'
+        m = re.match(slain_regexp, trunc_line)
+        if m:
+            # add this message to the list of death messages
+            # since this is just for testing, disarm the kill-gun
+            self._death_list.append(line)
+            starprint(f'DeathLoopVaccine:  Death count = {len(self._death_list)}')
+            self._kill_armed = False
+
+        # only do the list-purging if there are already some death messages in the list, else skip this
+        if len(self._death_list) > 0:
+
+            # create a datetime object for this line, using the very capable datetime.strptime()
+            now = datetime.strptime(line[0:26], '[%a %b %d %H:%M:%S %Y]')
+
+            # now purge any death messages that are too old
+            done = False
+            while not done:
```

Comment:
I'm still just scanning, but at first glance this doesn't look super Pythonic and I'm guessing this loop can be reformatted

Reply:
Sure.